internal/dag/builder.go

```diff
@@ -778,9 +778,8 @@ func (b *Builder) computeRoutes(sw *ObjectStatusWriter, proxy *projcontour.HTTPP
 				r.MirrorPolicy = &MirrorPolicy{
 					Cluster: c,
 				}
-			} else {
-				r.Clusters = append(r.Clusters, c)
 			}
+			r.Clusters = append(r.Clusters, c)
```

Comment:
I think that a better fix is to update `(*Route) Visit()` to visit the mirror policy as well as the direct route clusters. While you are there, could you also move it closer to the rest of the `Route` methods so that it's easier to find?

Reply:
So I had a think about this and actually did this in the cluster visit. I was brute-forcing this by addressing it in the route visit, but really we needed this in the cluster.visit.